### PR TITLE
Fix ndjson unset container encoding

### PR DESCRIPTION
**Pull request by henridf**

### CODE REVIEW

File: pkg/zsio/ndjson/ndjson_test.go [hunk 1]
```diff
@@ -7,11 +7,43 @@ import (
...
+`,
+			output: `{"dns":["google.com"],"email":null,"ip":null,"uri":null}`,
+		},
+		{
+			name: "nested nulls",
+			input: `#0:record[san:record[dns:vector[string],uri:vector[string],email:set[string],ip:vector[addr]]]
+0:[[[google.com;]-;-;-;]]
+`,
+			output: `{"san":{"dns":["google.com"],"email":null,"ip":null,"uri":null}}`,
+		},
```
Comment (mattnibs) [inline pkg/zsio/ndjson/ndjson_test.go]: Maybe include a test handles an empty vector or set?

Comment (henridf) [reply pkg/zsio/ndjson/ndjson_test.go]: Yes I realized after putting up this PR that there's a bug there too... will add and fix.

Comment (henridf) [reply pkg/zsio/ndjson/ndjson_test.go]: added tests for empties, will merge.

(turns out there was no bug there... my test was  wrong. )
